keyboards/tkc/tkl_ab87/keymaps/via/keymap.c

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2020 Terry Mathews
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+//Layers
+
+enum {
+  BASE = 0,
+  FUNCTION,
+  ALTERNATE,
+  LAST,
+};
```

Comment:
```suggestion
enum {
  BASE,
  FUNCTION,
};
```
4 layers for a TKL with 32u4 is **really** pushing it.
I'd suggest going with 2 or 3 layers.

Reply:
Reducing to two VIA keymaps took a few more changes than just this, but I pushed a commit that took care of it. FWIW, I tested the firmware on real hardware and it didn't crash.